plugins/reporters/opossum/src/main/kotlin/OpossumReporter.kt

```diff
@@ -96,50 +104,78 @@ class OpossumReporter(
     override val descriptor: PluginDescriptor = OpossumReporterFactory.descriptor,
     private val config: OpossumReporterConfig
 ) : Reporter {
+    @Serializable
     internal data class OpossumSignal(
-        val source: String,
-        val id: Identifier? = null,
-        val url: String? = null,
-        val license: SpdxExpression? = null,
-        val copyright: String? = null,
-        val comment: String? = null,
-        val preselected: Boolean = false,
-        val followUp: Boolean = false,
-        val excludeFromNotice: Boolean = false,
-        val uuid: UUID = UUID.randomUUID()
+        @Transient
```

Comment:
No, but it can be solved differently, I realized. I removed the property and collect the signals in a map `uuidToSignals` now. We could also collect them in a map `signalsToUUID` but that would require to replace the current `matches` method by overriding the `equals` and `hashCode` methods, I suppose. Would you prefer that? For now I didn't do it in the interest of keeping the diff smaller.

Reply:
> but that would require to replace the current `matches` method by overriding the `equals` and `hashCode` methods, I suppose. Would you prefer that?

I guess not. In Kotlin, I try to avoid overriding `equals` and `hashCode` as much as possible in favor of auto-generate code.

But couldn't `@Transient` then simply have been dropped and the original code be kept? Anyway, go for whatever causes less work for you.